app/models/event.rb

```diff
@@ -84,6 +84,7 @@ def self.events_to_calendar(format = nil)
     end
 
     events.each do |event|
+      next if event.is_provisional_date
```

Comment:
Instead of iterating through all the events and skipping those with provisional date, you could filter those out at the beginning with a `where` statement on the Even model

Reply:
Thanks @Levia I will change this!